File: JavaScript/Programmers/Level 1/크기가 작은 부분 문자열.js
// function getPartStrings(strings, targetLength) {
//     const result = []
//     for (let i = 0; i < [...strings].length; i++) {
//         if (strings.length - i < targetLength) return result
//         result.push(Number(strings.slice(i, i + targetLength)))
//     }
//     return result
// }

// function solution(t, p) {
//     const target = Number(p)
//     return getPartStrings(t, p.length).reduce((result, number) => result + Number(number <= target), 0)
// }

function solution(t, p) {
    let answer = 0
    for (let i = 0; i <= t.length - p.length; i++) {
        const part = Number(t.slice(i, i + p.length))
        if (part <= Number(p)) answer++
    }
    return answer
}

File: Kotlin/Programmers/Level 1/크기가 작은 부분 문자열.kt
// class Solution {
//     fun solution(t: String, p: String): Long {
//         var answer = 0L
//         for (i in 0..(t.length - p.length)) {
//             val current = t.slice(i until i + p.length).toLong()
//             if (current <= p.toLong()) answer += 1
//         }
//         return answer
//     }
// }

class Solution {
    fun solution(t: String, p: String): Int {
        return (0..(t.length - p.length))
            .map { t.slice(it until it + p.length) }
            .count { it <= p }
    }
}